### ISSUE TITLE
[Bug] Styles are not applied to shell

### ISSUE THREAD

**Issue by Mikilll94** [xamarin/Xamarin.Forms#7639]
### Description

I have created a simple Shell app based on VS2019 Shell template. In this template we have this section:

```xaml
    <Shell.Resources>
        <ResourceDictionary>
            <Color x:Key="NavigationPrimary">#FF0000</Color>
            <Style x:Key="BaseStyle" TargetType="Element">
                <Setter Property="Shell.TitleColor" Value="White" />
                <Setter Property="Shell.UnselectedColor" Value="Orange" />
                <Setter Property="Shell.TabBarBackgroundColor" Value="Yellow" />
                <Setter Property="Shell.TabBarForegroundColor" Value="Red"/>
                <Setter Property="Shell.TabBarUnselectedColor" Value="Brown"/>
                <Setter Property="Shell.TabBarTitleColor" Value="Green"/>
            </Style>
            <Style TargetType="ShellItem" BasedOn="{StaticResource BaseStyle}" />
        </ResourceDictionary>
    </Shell.Resources>
```

However changing these styles does not bring any effect to application.

I.e. changing TabBarUnselectedColor does not change unselected tab colors in "Cars" flyout item.

### Steps to Reproduce

1. Run attached project
2. Change any colors in Shell.Resources section
3. Nothing happens. Changes are not applied.

### Expected Behavior

Styling shell should be possible.

### Actual Behavior

Shell cannot be styled.

### Basic Information

- Version with issue: 4.2.0.815419
- IDE: VS 2019 16.2.4
- Affected Devices:

### Reproduction Link

[ShellTest.zip](https://github.com/xamarin/Xamarin.Forms/files/3643010/ShellTest.zip)

**Comment by VladislavAntonyuk** [comment]
you have to apply this style to your shell
```xml

<Shell
    ItemTemplate="{StaticResource FlyoutItemTemplate}"
    MenuItemTemplate="{StaticResource MenuTemplate}"
    Style="{StaticResource BaseStyle}">

    <Shell.Resources>
        <ResourceDictionary>

            <Style x:Key="BaseStyle" TargetType="Element">
                <Setter Property="Shell.TitleColor" Value="White" />
                <Setter Property="Shell.UnselectedColor" Value="Orange" />
                <Setter Property="Shell.TabBarBackgroundColor" Value="Yellow" />
                <Setter Property="Shell.TabBarForegroundColor" Value="Red"/>
                <Setter Property="Shell.TabBarUnselectedColor" Value="Brown"/>
                <Setter Property="Shell.TabBarTitleColor" Value="Green"/>
            </Style>
            <Style BasedOn="{StaticResource BaseStyle}" TargetType="ShellItem" />
            <Style BasedOn="{StaticResource BaseStyle}" TargetType="FlyoutItem" />
```

**Comment by Mikilll94** [comment]
@VladislavAntonyuk 
Your solution is a little bit redundant, i.e. this line
```
Style="{StaticResource BaseStyle}"
```
is completely unnecessary.


I have checked my example once again and I have found a solution.

Just add this line to my project:
```
<Style TargetType="Tab" BasedOn="{StaticResource BaseStyle}" />
```
and now styling tabs work fine. IMO styling Shell should be better documented in docs with some examples.

@hartez 
I think this issue can be closed, because I have found out why this was not working.
